### PR TITLE
TLS 1.3 PSK: add option to require only PSK with DHE

### PR DESCRIPTION
**Pull request by SparkiDev**
# Description

Can specify only PSK without DHE.
Add only PSK with DHE.

Fixes zd#15456

# Testing
Regression tested TLS 1.3

# Checklist

 - [x] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

File: src/tls.c [hunk 1]
```diff
@@ -12137,16 +12137,25 @@ int TLSX_PopulateExtensions(WOLFSSL* ssl, byte isServer)
...
+                }
             #if !defined(NO_DH) || defined(HAVE_ECC) || \
                               defined(HAVE_CURVE25519) || defined(HAVE_CURVE448)
-                if (!ssl->options.noPskDheKe)
+fprintf(stderr, "NO PSK DHE KE: %d\n", ssl->options.noPskDheKe);
+                if (!ssl->options.noPskDheKe) {
                     modes |= 1 << PSK_DHE_KE;
+                }
             #endif
+fprintf(stderr, "MODES: %d\n", modes);
```
Comment (dgarske) [inline src/tls.c:12158]: Did you mean for this to get included in the PR?

File: src/tls.c [hunk 2]
```diff
@@ -12137,16 +12137,25 @@ int TLSX_PopulateExtensions(WOLFSSL* ssl, byte isServer)
...
             #endif
             {
-                byte modes;
+                byte modes = 0;
 
                 (void)usingPSK;
                 /* Pre-shared key modes: mandatory extension for resumption. */
-                modes = 1 << PSK_KE;
+            #ifdef HAVE_SUPPORTED_CURVES
+fprintf(stderr, "ONLY PSK DHE KE: %d\n", ssl->options.onlyPskDheKe);
```
Comment (dgarske) [inline src/tls.c:12145]: Did you mean for this to get included in the PR?

File: src/tls13.c [hunk 3]
```diff
@@ -11917,6 +11924,38 @@ int wolfSSL_no_dhe_psk(WOLFSSL* ssl)
     return 0;
 }
 
+#ifdef HAVE_SUPPORTED_CURVES
```
Comment (dgarske) [inline src/tls13.c:11927]: Should these also be wrapped with `!defined(NO_PSK)`?

**Review by dgarske — CHANGES_REQUESTED** [review]

File: src/tls.c (hunk 1 above)
Comment (SparkiDev) [reply src/tls.c:12158]: Removed

File: src/tls.c (hunk 2 above)
Comment (SparkiDev) [reply src/tls.c:12145]: Removed

File: src/tls13.c (hunk 3 above)
Comment (SparkiDev) [reply src/tls13.c:11927]: OK.
TLS 1.3 PSK is PSK and session tickets.
Added protection around fields and code.

**Review by dgarske — APPROVED** [review]

**Pull request merged**